### ISSUE TITLE
Feature Request:- Add Sub-Task. 

### ISSUE THREAD

**Issue by Salkalkhan** [maxpozdeev/mytinytodo#37]
I think Instead Of **Markdown,** their Should be **_Json_** based Sub-Task System. Where We can Add , Sort, insert Sub-Sub-task and delete task And Json Data in Db will only be affected.

**Comment by indrakurnia** [comment]
Vote for sub-tasks feature...

**Comment by maxpozdeev** [comment]
I do not agree to use json in task note to insert sub tasks. I think sub-task should be the same as a usual task with tags and duedate, but another level of visibility (they are not displayed by default). As well sub-tasks should be found if we use filters like tags or duedate (probably a checkbox to filter all subtasks or only top-level tasks), this is a subject to consider.

Example. When we open a list we see top-level tasks with a note that some tasks has a number of subtasks. If we click on this note we see a list of sub-tasks.

<img width="557" alt="subtasks-idea" src="https://user-images.githubusercontent.com/1649326/196445870-aba468e7-7b2a-40e6-8a7a-eccd63e6d115.png">

**Comment by devgaucho** [comment]
> I think Instead Of **Markdown,** their Should be **_Json_** based Sub-Task System. Where We can Add , Sort, insert Sub-Sub-task and delete task And Json Data in Db will only be affected.

instead of removing markdown you could simply add support for clickable checkboxes to it

ex:

```markdown
- [x] item
```

clickable checkboxes are already part of the markdown used on github for example, the only difference would be that they would be interactive and when clicked would change the markdown code like in a wysiwyg editor

**Comment by DHS1001** [comment]
it would be great to have the sub-tasks feature in the future release.
